testers/markus_tester.py

```diff
@@ -343,26 +342,54 @@ def upload_svn_feedback(self, markus_root_url, repo_name, assignment_name, svn_f
                           repo_path]
         subprocess.run(svn_ci_command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
 
+    def before_tester_run(self):
+        """
+        Callback invoked before running this tester.
+        Use this for tester initialization steps that can fail, rather than using __init__.
+        """
+        pass
+
     def before_test_run(self, test):
         """
-        Callback method invoked after initializing each test, but before running them.
-        :param test: The test about to be executed.
+        Callback invoked before running a test.
+        Use this for test initialization steps that can fail, rather than using test_class.__init__().
+        :param test: The test after initialization.
         """
         pass
 
+    def get_custom_test_arguments(self):
+        """
+        Gets a dict of custom arguments to be passed to the test constructor.
+        :return: The dict of custom arguments.
+        """
+        return {}
+
     def after_test_run(self, test):
```

Comment:
I think this should be renamed to `after_successful_test_run` to make it clear that this should not be used for cleanup code. This will distinguish it from `after_tester_run` which **is** run regardless of a success or failure. 
We could even introduce an `after_failed_test_run` as well.